tests/test_bgp_af.rb

```diff
@@ -563,122 +404,79 @@ def test_dampening
     assert_equal(bgp_af.dampening_routemap, 'DropAllTraffic',
                  'Error: dampening getter did not match')
 
-    #
-    # Unset and verify
-    #
     bgp_af.dampening = nil
-    af_string = get_bgp_af_cfg(asn, vrf, af)
-    pattern = /^ *dampening$/
-
-    refute_match(pattern, af_string, "Error: 'dampening' is still configured")
+    assert_nil(bgp_af.dampening)
+    assert_nil(bgp_af.dampening_half_time)
+    assert_nil(bgp_af.dampening_reuse_time)
+    assert_nil(bgp_af.dampening_suppress_time)
+    assert_nil(bgp_af.dampening_max_suppress_time)
+    assert_nil(bgp_af.dampening_routemap)
 
     #############################################
     # Set and verify 'dampening' with default   #
     #############################################
-    vrf = 'sangria'
-    af = %w(ipv4 multicast)
-    bgp_af = RouterBgpAF.new(asn, vrf, af)
-
-    bgp_af.dampening = bgp_af.default_dampening
-
-    # Check property got set
-    af_string = get_bgp_af_cfg(asn, vrf, af)
-
-    assert_match(pattern, af_string,
-                 "Error: 'dampening' is not configured and should be")
 
-    # Check properties got set
-    af_string = get_bgp_af_dampening_params(asn, vrf, af)
-
-    pattern_params1 = 'Half-life time                 : 15 mins'
-    pattern_params2 = 'Suppress penalty               : 2000'
-    pattern_params3 = 'Reuse penalty                  : 750'
-    pattern_params4 = 'Max suppress time              : 45 mins'
-    pattern_params5 = 'Max suppress penalty           : 6000'
-
-    error = ("Error: 'dampening' properties are incorrect")
-
-    assert_match(pattern_params1, af_string, error)
-    assert_match(pattern_params2, af_string, error)
-    assert_match(pattern_params3, af_string, error)
-    assert_match(pattern_params4, af_string, error)
-    assert_match(pattern_params5, af_string, error)
-
-    # Check getters
-    assert_empty(bgp_af.dampening, 'Error: dampening not configured ' \
-                 'and should be')
-    assert_equal(bgp_af.default_dampening_half_time, bgp_af.dampening_half_time,
-                 'Wrong default dampening half_time value configured')
-    assert_equal(bgp_af.default_dampening_reuse_time,
-                 bgp_af.dampening_reuse_time,
-                 'Wrong default dampening reuse_time value configured')
-    assert_equal(bgp_af.default_dampening_suppress_time,
-                 bgp_af.dampening_suppress_time,
-                 'Wrong default dampening suppress_time value configured')
-    assert_equal(bgp_af.default_dampening_max_suppress_time,
-                 bgp_af.dampening_max_suppress_time,
-                 'Wrong default dampening suppress_max_time value configured')
-    assert_equal(bgp_af.default_dampening_routemap,
-                 bgp_af.dampening_routemap,
-                 'The default dampening routemap should configured')
+    # IOS XR; we skip this section because we cannot query
+    #    the default parameters
```

Comment:
Charles an I looked at this one before I understood the default_values / auto_default semantics.  I think I could fix it now, but would prefer to push to US59615.